src/socket/communicate.rs

```diff
@@ -0,0 +1,239 @@
+//! Communication with the lorri daemon.
+//!
+//! We provide a fixed number of `CommunicationType`s that are sent
+//! between the lorri daemon and lorri clients wishing to connect to it.
+//!
+//! `listener` implements the daemon side, which provides an `accept`
+//! method which is similar to how `accept()` works on a plain Unix socket.
+//!
+//! `client` implements a set of clients specialized to the communications
+//! we support.
+
+use std::os::unix::net::UnixStream;
+
+use crate::socket::path::{BindError, BindLock, SocketPath};
+use crate::socket::{ReadWriteError, ReadWriter, Timeout};
+use crate::NixFile;
+
+/// We declare 1s as the time readers should wait
+/// for the other side to send something.
+pub const DEFAULT_READ_TIMEOUT: Timeout = Timeout::from_millis(1000);
+
+/// Enum of all communication modes the lorri daemon supports.
+#[derive(Serialize, Deserialize)]
+pub enum CommunicationType {
+    /// Ping the daemon from a project to tell it to watch & evaluate
+    // TODO: rename to BuildProject or something?
+    Ping,
+    // /// Listen for events on the daemon
```

Comment:
```suggestion
```

Reply:
Should we keep that around as a template for implementing more complex server messages?

Reply:
I don't think so, personally. I'd rather avoid keeping comments around